### ISSUE TITLE
Do not use deprecated methods

### ISSUE THREAD

**Issue by mmmsharap** [jbox-web/ajax-datatables-rails#143]
Method [each_value](https://github.com/antillas21/ajax-datatables-rails/blob/master/lib/ajax-datatables-rails/base.rb#L77) is deprecated and will be removed in Rails 5.1, as `ActionController::Parameters` no longer inherits from hash. Using this deprecated behavior exposes potential security problems. If you continue to use this method you may be creating a security vulnerability in your app that can be exploited. Instead, consider using one of [these documented methods](http://api.rubyonrails.org/v5.0.0.beta3/classes/ActionController/Parameters.html.) which are not deprecated

**Comment by ajahongir** [comment]
@mmmsharap try to use latest version - [v-0-4-0](https://github.com/antillas21/ajax-datatables-rails/tree/v-0-4-0)

**Comment by ghost** [comment]
Still getting these 2 deprecation warnings on v-0-4-0:

```
DEPRECATION WARNING: Method map is deprecated and will be removed in Rails 5.1, as `ActionController::Parameters` no longer inherits from hash. Using this deprecated behavior exposes potential security problems. If you continue to use this method you may be creating a security vulnerability in your app that can be exploited. Instead, consider using one of these documented methods which are not deprecated: http://api.rubyonrails.org/v5.0.0.1/classes/ActionController/Parameters.html (called from columns at D:/Ruby231/lib/ruby/gems/2.3.0/bundler/gems/ajax-datatables-rails-a5950cc14da7/lib/ajax-datatables-rails/datatable/datatable.rb:41)
DEPRECATION WARNING: Method map is deprecated and will be removed in Rails 5.1, as `ActionController::Parameters` no longer inherits from hash. Using this deprecated behavior exposes potential security problems. If you continue to use this method you may be creating a security vulnerability in your app that can be exploited. Instead, consider using one of these documented methods which are not deprecated: http://api.rubyonrails.org/v5.0.0.1/classes/ActionController/Parameters.html (called from orders at D:/Ruby231/lib/ruby/gems/2.3.0/bundler/gems/ajax-datatables-rails-a5950cc14da7/lib/ajax-datatables-rails/datatable/datatable.rb:21)
```

**Comment by Praveen-Invoscape** [comment]
I am also have the same issue..is it any ways to solve that

**Comment by ReckerSwartz** [comment]
+1

**Comment by reckerswartz** [comment]
same issue

**Comment by jmaggesi** [comment]
Same issue here.

I suspect that to solve the issue we should **`.permit`** the parameters and then call a **`.to_h`** to have the Hash and do the **`.each_value`** call.

[Look Here](http://stackoverflow.com/questions/34949505/rails-5-unable-to-retrieve-hash-values-from-parameter)

**Comment by godinezb** [comment]
Is this already solved?

**Comment by ajahongir** [comment]
https://github.com/antillas21/ajax-datatables-rails/issues/143

**Comment by ajahongir** [comment]
https://github.com/antillas21/ajax-datatables-rails/issues/194

**Comment by SampsonCrowley** [comment]
@cdekker this is fixed in pull request #208 in the v4 banch

**Comment by matissg** [comment]
If anyone wants to keep using **ajax-datatables-rails v.0.3.1** on Rails 5 here is fix for that - [https://github.com/antillas21/ajax-datatables-rails/pull/198](url)
Basically it changes `each_value` to `each_pair` and then it works like charm. Just tested on Rails 5.1.
I created clone with this fix included here - [https://github.com/matissg/ajax-datatables-rails-v-0-3-Rails-5-1](url)

**Comment by leandromarques** [comment]
@matissg your solution isn't avaiable anymore.

**Comment by leandromarques** [comment]
Well, I've fixed the problem on my own for Rails 5.1. I don't tested in another Rails versions. So, if anyone wants improve this and do a pull request, here is.

https://gist.github.com/leandromarques/86fac39095384cd8d2d9d3a35ffbd9cf

**Comment by n-rodriguez** [comment]
Fixed: https://github.com/jbox-web/ajax-datatables-rails/commit/7e92f7b3e635d30ad096772cdb25e1eb0b552fa5
